tests/tracer/test_processors.py

```diff
@@ -134,11 +134,45 @@ def test_aggregator_reset_default_args():
     assert dd_proc in aggr.dd_processors
     assert user_proc in aggr.user_processors
     assert aggr.writer is not dm_writer
-    assert sampling_proc is not aggr.sampling_processor
+    assert sampling_proc is aggr.sampling_processor
     assert not aggr._traces
     assert len(aggr._span_metrics["spans_created"]) == 0
 
 
+def test_aggregator_reset_apm_opt_out_preserves_sampling():
+    """
+    Test that calling aggr.reset(apm_opt_out=True) updates the apm_opt_out setting
+    but preserves the sampling rules on the TraceSamplingProcessor.
+    """
+    from ddtrace.internal.sampling import SpanSamplingRule
+
+    sampling_rule = SpanSamplingRule(service="test_service", name="test_name", sample_rate=0.5, max_per_second=10)
+
+    dd_proc = DummyProcessor()
+    user_proc = DummyProcessor()
+    aggr = SpanAggregator(
+        partial_flush_enabled=False,
+        partial_flush_min_spans=1,
+        dd_processors=[dd_proc],
+        user_processors=[user_proc],
+    )
+
+    sampling_proc = aggr.sampling_processor
+    original_apm_opt_out = sampling_proc.apm_opt_out
+
+    sampling_proc.single_span_rules = [sampling_rule]
```

Comment:
Can we also set a SamplingRule (trace sampling rule not single span sampling). That's the key thing we need to test here.